core/src/main/scala/persistentscheduler/PersistentScheduler.scala

```diff
@@ -63,12 +68,11 @@ class PersistentScheduler(persistence: SchedulerPersistence) extends Actor
       this.subscriptions = subscriptions;
   }
 
-  def addSubscription(subscription: Subscription): Unit = {
+  def addSubscription(subscription: Subscription): Future[Any] = {
     subscriptions = subscriptions + subscription
-    sender() ! SubscribedActorRef(sender())
+    sender() ? SubscribedActorRef(sender())
```

Comment:
here as well and all the other places in this method, you don't need the ask pattern here